src/index.ts

```diff
@@ -66,6 +67,18 @@ export default function electron(options: ElectronOptions | ElectronOptions[]):
               {
                 name: ':startup',
                 closeBundle() {
+                  /*
+                  closeBundle is emitted for both the preload and main electron
+                  files. If we simply run Electron once either one finishes,
+                  the other one might not finish in time. Usually, the faster
+                  one is the preload, meaning if main is *a lot* larger than it,
+                  Electron might start before the file is done building,
+                  leading to an "Unable to find Electron app at ..." error
+                   */
+                  closeBundleCount++
+                  if (closeBundleCount < optionsArray.length)
```

Comment:
It seems that we need to judge based on the number of entry files, so the judgment conditions here are not accurate.
It may be possible to detect how many entry files there are in the `configResolved` hook, if that is possible I will consider merging this PR.

**e.g.**

```js
import electron from 'vite-plugin-electron'
import simpleElectron from 'vite-plugin-electron/simple'

export default {
  plugins: [
    // Case 1
    electron([
      {
        entry: {
          foo: 'foo.ts',
          bar: 'bar.ts',
        },
      },
    ]),

    // Case 2
    electron({
      entry: {
        foo: 'foo.ts',
        bar: 'bar.ts',
      },
    }),

    // Case 3
    electron({
      entry: [
        'foo.ts',
        'bar.ts',
      ],
    }),

    // Case: use the simple API
    simpleElectron(/* Same as above */),
  ],
}
```

Reply:
Let's see here:
- For each option object provided, we wait for one event & call Vite's `build` function *once*
- Vite calls Rollup's `bundle.write` or `bundle.generate` functions (see [source code for Vite](https://github.com/vitejs/vite/blob/cb3ab5f465e0de340c4d8b82603e7ee1489b5b62/packages/vite/src/node/build.ts#L673))
- According to [Rollup's Plugin development docs](https://rollupjs.org/plugin-development/#output-generation-hooks), *one* call to write/generate will result in exactly *one* `closeBundle` event

TLDR: The current implementation works